packages/agoric-cli/src/lib/yarn.js

```diff
@@ -0,0 +1,20 @@
+// @ts-check
+import { execFileSync } from 'child_process';
+
+/**
+ * Omits the root
+ *
+ * @returns {Array<{ location: string, name: string }>}
+ */
+export const listWorkspaces = () => {
+  const ndjson = execFileSync('yarn', ['workspaces', 'list', '--json'], {
```

Comment:
exported `listWorkspaces` closes over ambient `execFileSync`.

How about we make it possible for the caller to pass the authority explicitly?

```suggestion
import { execFileSync as execFileSyncAmbient } from 'child_process';

/**
 * Omits the root
 *
 * @param {{ execFileSync?: typeof execFileSyncAmbient }} [io]
 * @returns {Array<{ location: string, name: string }>}
 */
export const listWorkspaces = (io = {}) => {
  const { execFileSync = execFileSyncAmbient } = io;
```

Reply:
ok. Pushed into a8ff178f144fdc197277f80530615d8144193fa3